Framework/PythonInterface/mantid/api/src/Exports/SpectrumInfo.cpp

```diff
@@ -1,9 +1,28 @@
 #include "MantidAPI/SpectrumInfo.h"
+#include "MantidTypes/SpectrumDefinition.h"
+
 #include <boost/python/class.hpp>
+#include <boost/python/copy_const_reference.hpp>
+#include <boost/python/list.hpp>
+#include <boost/python/return_value_policy.hpp>
 
+using Mantid::SpectrumDefinition;
 using Mantid::API::SpectrumInfo;
 using namespace boost::python;
 
+// Helper function to create and return a list of all the spectrum definitions
+boost::python::list getSpectrumDefinitionList(const SpectrumInfo &self) {
+  // Create the list
+  boost::python::list spectrumDefinitionList;
+
+  // Get all the spectrum definitions
+  for (auto i = 0; i < self.size(); i++) {
+    spectrumDefinitionList.append(self.spectrumDefinition(i));
+  }
```

Comment:
I like this, but we should also check the performance. If it is slower then you might need to ask around to check what the largest realistic grouping used is (we do not use anything particularly large at the ILL).